### PR TITLE
Refactor function skip loop in `getRightThis()` to a separate function

### PR DESCRIPTION
**Pull request by SSoulaimane**
Fragment from #9282

### CODE REVIEW

File: src/dmd/expressionsem.d [hunk 1]
```diff
@@ -11508,3 +11484,55 @@ private bool checkFunctionAttributes(Expression exp, Scope* sc, FuncDeclaration
...
+ *      s = the parent symbol of the existing `this`
+ *      ad = struct or class we need the correct `this` for
+ *      e1 = existing `this`
+ *      t = type of the existing `this`
+ *      var = the specific member of ad we're accessing
+ *      flag = if true, return `null` instead of throwing an error
+ * Returns:
+ *      Expression representing the `this` for the var
+ */
+Expression skipNestedFunctions(const ref Loc loc, Scope* sc, Dsymbol s, AggregateDeclaration ad, Expression e1, Type t, Dsymbol var, bool flag = false)
```
Comment (JinShil) [inline src/dmd/expressionsem.d]: The fact that this name does not have `getThis` in it makes it less obvious when trying to comprehend the usage sites.  What about `getThisSkippingNested` or something like that?

File: src/dmd/expressionsem.d [hunk 2]
```diff
@@ -11508,3 +11484,55 @@ private bool checkFunctionAttributes(Expression exp, Scope* sc, FuncDeclaration
...
+ * Gets `this` of the next outer aggregate.
+ * Params:
+ *      loc = location to use for error messages
+ *      sc = context
+ *      s = the parent symbol of the existing `this`
+ *      ad = struct or class we need the correct `this` for
+ *      e1 = existing `this`
+ *      t = type of the existing `this`
+ *      var = the specific member of ad we're accessing
+ *      flag = if true, return `null` instead of throwing an error
```
Comment (JinShil) [inline src/dmd/expressionsem.d]: I think `flag` could have been omitted from this PR, but I see where this is going.

**Review by JinShil — APPROVED** [review]
Aside from the name, looks ok to me

File: src/dmd/expressionsem.d (hunk 1 above)
Comment (SSoulaimane) [reply src/dmd/expressionsem.d]: agreed

File: src/dmd/expressionsem.d (hunk 2 above)
Comment (SSoulaimane) [reply src/dmd/expressionsem.d]: why do you thinks so? it is used to keep the current behavior untouched. in `expressionsem.d` an error message is generated but not in `typesem.d`.

File: src/dmd/expressionsem.d (hunk 1 above)
Comment (SSoulaimane) [reply src/dmd/expressionsem.d]: renmed to `getThisSkipNestedFuncs`.

File: src/dmd/expressionsem.d (hunk 2 above)
Comment (JinShil) [reply src/dmd/expressionsem.d]: Ah, you are right.  My apologies.

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**